### PR TITLE
Update lobster version log 

### LINKED ISSUE #433: Log state of Lobster software
It would be good to have a way to somehow log the state of the Lobster software-- preferably the hash of HEAD. It's hard to debug when that info is lost.

### PR DESCRIPTION
**Pull request by WenzhaoLi**
Update the Lobster version representation.
For example: 2.1-a234fea-dirty (the last part marking uncommitted in changes)
Also print the Lobster version in process.log.

close #433

Also replace some special symbols, which cause weird characters for Putty users.
Make it more Windows friendly.

### CODE REVIEW

File: lobster/util.py [hunk 1]
```diff
@@ -312,8 +313,15 @@ def verify(workdir):
     if not os.path.exists(workdir):
         return
 
-    my_version = get_distribution('Lobster').version
+    my_version = get_version()
+    #my_version = get_distribution('Lobster').version
```
Comment (annawoodard) [inline lobster/util.py]: I'd recommend removing this.

File: setup.py [hunk 2]
```diff
@@ -1,10 +1,16 @@
 #!/usr/bin/env python
 
 from setuptools import setup
+import subprocess
+import shlex
+
+head = subprocess.check_output(shlex.split('git rev-parse --short HEAD')).strip()
```
Comment (annawoodard) [inline setup.py]: Why not just re-use the function you added to `util.py`? This is exactly the same.

File: lobster/util.py [hunk 3]
```diff
@@ -312,8 +313,15 @@ def verify(workdir):
...
+    major,  head, status = my_version.split('-')
+    my_version = major
+
     stored_version = checkpoint(workdir, 'version')
+    major,  head, status = stored_version.split('-')
+    stored_version = major
+
```
Comment (annawoodard) [inline lobster/util.py]: This is very minor-- I'm only mentioning it FYI. It's conventional in Python to use an underscore to denote a variable that you aren't going to use. That signals to the person trying to understand the code: don't spend any time figuring out what this variable is, it doesn't get used anywhere. In my opinion it makes code a bit more readable, but to each their own. So the lines above could look something like:
```
current_major, _, _ = get_version().split('-')
stored_major, _, _ = checkpoint(workdir, 'version').split('-')
if current_major != stored_major:
    raise ValueError("Lobster {0!r} cannot process a run created with version {1!r}".format(
        my_version, stored_version))
```

**Comment by annawoodard** [comment]
This looks good to me! I just had some small comments. Good job Wenzhao!! 👏 👏 👏 

Now let's wait for feedback from @matz-e. He might have a better idea than changing directories like I suggested, which is kind of kludgy. Note @matz-e that `get_version()` works whether you are running `lobster` after `pip install` or running `python lobster.py`.

File: lobster/util.py [hunk 4]
```diff
@@ -278,7 +278,7 @@ def id2dir(id):
     # Currently known limitations on the number of entries in a
     # sub-directory concern ext3, where said limit is 32k.  Use a
     # modus of 10k to split the task numbers.  Famous last words:
-    # "(10k)Â² tasks should be enough for everyone." we use two levels
+    # "(10k)squared tasks should be enough for everyone." ->we use two levels
```
Comment (matz-e) [inline lobster/util.py]: This looks fine in the browser to me — [like here](https://github.com/matz-e/lobster/blob/master/lobster/util.py#L280). If you have to change this, please use `^2`. Further, `putty` seems a very dated way to log in somewhere to me by now. UTF-8 has been a standard/thing for quite a while. Maybe the anniversary update for Windows 10 will make this a little better — I'll give it a try if my machine at home has already updated.

File: lobster/util.py [hunk 5]
```diff
@@ -7,6 +7,7 @@
 import shutil
 import subprocess
 import yaml
+import shlex
```
Comment (matz-e) [inline lobster/util.py]: I think that imports should be sorted alphabetically — that way, it is very easy to see if something has been imported already.

File: lobster/util.py (hunk 3 above)
Comment (matz-e) [inline lobster/util.py]: Seconded.

File: setup.py (hunk 2 above)
Comment (matz-e) [inline setup.py]: I concur, again.

File: lobster/util.py (hunk 1 above)
Comment (matz-e) [inline lobster/util.py]: Adding a little: when you comment out code you should always give a reason or explanation. I've wasted long hours of time tripping over something commented out without further information and then going back through source code history trying to figure out *why* this code was changed, and if the old version is somehow important to me.

This is one of the bad practices common in HEP — commented out code should be dead code and removed.

**Comment by matz-e** [comment]
@WenzhaoLi: Thank you! I think this looks good once comments are addressed. I think @annawoodard is spot on with her remarks.

I don't see a simpler version of `get_version` right now. I don't think it matters, either. It's readable, and pretty compact.

File: setup.py (hunk 2 above)
Comment (WenzhaoLi) [inline setup.py]: I thought the `get_distribution()` can only get the message in `setup.py`. So is there any other way to know the installed lobster version? (except `get_distribution()`)

**Comment by WenzhaoLi** [comment]
Thanks! 
Should I undo the commit or make a new commit for the change?

**Comment by annawoodard** [comment]
Make a new commit.

**Comment by annawoodard** [comment]
Then you can push your branch again, it will update the PR.

**Comment by WenzhaoLi** [comment]
I tried importing `util.py` to `setup.py`. But it didn't work. 
```
Processing /afs/crc.nd.edu/user/w/wli12/lobster
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-h_D5Jr-build/setup.py", line 4, in <module>
        from lobster.util import get_version
      File "lobster/__init__.py", line 1, in <module>
        import cmssw
      File "lobster/cmssw/__init__.py", line 1, in <module>
        import dash
      File "lobster/cmssw/dash.py", line 10, in <module>
        from WMCore.Services.SiteDB.SiteDB import SiteDBJSON
      File "/afs/crc.nd.edu/user/w/wli12/.lobster/lib/python2.7/site-packages/WMCore/Services/SiteDB/SiteDB.py", line 8, in <module>
        from WMCore.Services.Service import Service
      File "/afs/crc.nd.edu/user/w/wli12/.lobster/lib/python2.7/site-packages/WMCore/Services/Service.py", line 89, in <module>
        from httplib2 import HttpLib2Error
    ImportError: No module named httplib2
```

**Comment by matz-e** [comment]
You can try to remove the `import cmssw` from `__init__.py` in `lobster`.  If you get errors executing Lobster, try changing import statements from `from lobster import cmssw` to `import lobster.cmssw as cmssw`?

I think it would be good to de-couple this so that we can import the utility functions without additional packages.

**Pull request merged**